lib/mongo/cluster.rb

```diff
@@ -194,10 +198,10 @@ def initialize(seeds, monitoring, options = Options::Redacted.new)
     # @return [ Proc ] The Finalizer.
     #
     # @since 2.2.0
-    def self.finalize(pools, cursor_reaper)
+    def self.finalize(pools, periodic_executor)
       proc do
-        begin; cursor_reaper.kill_cursors; rescue; end
-        cursor_reaper.stop!
+        periodic_executor.stop!
+        begin; periodic_executor.execute; rescue; end
```

Comment:
Good catch, thanks!
I changed the #stop! method on the periodic executor to only stop the thread and created another method, #flush that calls #flush on the executors to execute pending operations. What do you think?